plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/TokenCredentialType.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.azure;
+
+// Type of token credentials that the plugin supports
+public enum TokenCredentialType {
+    MANAGED_IDENTITY
```

Comment:
You could make the setting convenient by supporting alternative string representation,  so users would have convenient value `token_credential_type: managed`:

```
public enum TokenCredentialType {
    MANAGED_IDENTITY("managed");
    
    private final String type;
    
    private TokenCredentialType(String type) {
        this.type = type;
    }
    
    static TokenCredentialType of(String type) {
        for (TokenCredentialType value: values()) {
            if (value.type.equalsIgnoreCase(type)) {
                return value;
            }
        }

        throw new IllegalArgumentException( );
    }
}
```

Reply:
refactored